src/main/java/by/yahor_kulesh/dao/TicketDAO.java

```diff
@@ -0,0 +1,119 @@
+package by.yahor_kulesh.dao;
+
+import by.yahor_kulesh.model.tickets.BusTicket;
+import by.yahor_kulesh.model.tickets.ConcertTicket;
+import by.yahor_kulesh.model.tickets.Ticket;
+import by.yahor_kulesh.utils.ObjectArray;
+
+import java.sql.*;
+import java.time.ZoneId;
+import java.util.UUID;
+
+public class TicketDAO{
+    private ResultSet rs = null;
+
+    public void insert(Ticket ticket){
+        try(
+                Connection con = DriverManager.getConnection("jdbc:postgresql://localhost:5432/my_ticket_service_db","postgres","postgres");
+                PreparedStatement stat = con.prepareStatement("INSERT INTO ticket(id,user_id,ticket_type,creation_date) VALUES (?,?,CAST(? AS ticket_type),?)")
+        ){
+            prepareTicketForStatement(ticket, stat);
+            stat.execute();
+        }catch(SQLException e){
+            System.err.println(e.getMessage());
+        }
+    }
+
+    public Ticket getById(UUID id) {
+        try{
+            try(
+                    Connection con = DriverManager.getConnection("jdbc:postgresql://localhost:5432/my_ticket_service_db", "postgres", "postgres");
+                    PreparedStatement stat = con.prepareStatement("SELECT * FROM ticket where id=?")
+            ) {
+                stat.setObject(1, id);
+                rs = stat.executeQuery();
+                if(rs.next()){
+                    return getTicketFromDB();
+                }else return null;
+            } finally {
+                closeResultSet();
+            }
+        } catch(SQLException e) {
+            System.err.println(e.getMessage());
+            return null;
+        }
+    }
+
+    public ObjectArray getByUserId(UUID id) {
+        ObjectArray tickets = new ObjectArray();
+        try{
+            try(
+                    Connection con = DriverManager.getConnection("jdbc:postgresql://localhost:5432/my_ticket_service_db", "postgres", "postgres");
+                    PreparedStatement stat = con.prepareStatement("SELECT * FROM ticket where user_id=?")
+            ) {
+                stat.setObject(1, id);
+                rs = stat.executeQuery();
+                while(rs.next()) {
+                    tickets.add(getTicketFromDB());
+                }
+                return tickets;
+            } finally {
+                closeResultSet();
```

Comment:
When a Statement object is closed, its current ResultSet object, if one exists, is also closed. You so need to close it additionally explicitly

Reply:
I close ResultSet  in closeResultSet() method

Reply:
you have answered at lection about it
solved